### PR TITLE
Add support on num steps for learning rate scheduler

### PR DESCRIPTION
**Pull request by sichu2023**
```bash
python sub-packages/bionemo-esm2/src/bionemo/esm2/scripts/train_esm2.py ... --learning-rate-num-steps=...
```

### CODE REVIEW

**Comment by sichu2023** [comment]
@skothenhill-nv Do you know where I should update `TrainingConfig` to mirror the change in `train.py`?
https://github.com/NVIDIA/bionemo-framework/blob/sichu/esm2_learning_rate_num_steps/sub-packages/bionemo-llm/src/bionemo/llm/train.py#L216

**Review by dorotat-nv — APPROVED** [review]

**Review by pstjohn — APPROVED** [review]

**Review by malcolmgreaves — APPROVED** [review]
